### ISSUE TITLE
strict_gt=True doesn't affect phased genotypes

### ISSUE THREAD

**Issue by davmlaw** [brentp/cyvcf2#198]
strict_gt=True works with unphased genotypes (ie "0/." => Unknown) but it leaves "0|." as HET

Test program:

```
import cyvcf2
for variant in cyvcf2.VCF("./cyvcf2_phased_strict_gt.vcf", strict_gt=True):
    print(variant)
    print(variant.gt_types)
```

Test VCF (cyvcf2_phased_strict_gt.vcf):

```
##fileformat=VCFv4.2
##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">
##contig=<ID=chr1,length=248956422>
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	sample_1
chr1	231373681	rs560634812	T	C	12816.8	PASS	.	GT	0/.
chr1	231373681	rs560634812	T	C	12816.8	PASS	.	GT	0|.
```

Output:

```
chr1	231373681	rs560634812	T	C	12816.8	PASS	.	GT	0/.

[2]
chr1	231373681	rs560634812	T	C	12816.8	PASS	.	GT	0|.

[1]
```

Expected output:

The 2nd VCF entry (GT="0|.") would also return gt type as "2" ie unknown not HET

**Comment by davmlaw** [comment]
PS, I wrote a workaround and haven't benchmarked it in Cython but maybe this implementation is faster than using range - "count_nonzero" which counts how many are True:

```
num_hom_ref = np.count_nonzero(gt_types == 0)
num_het = np.count_nonzero(gt_types == 1)
num_unknown = np.count_nonzero(gt_types == (3 if self.vcf.gts012 else 2))
num_hom_alt = np.count_nonzero(gt_types == (2 if self.vcf.gts012 else 3))
```

**Comment by brentp** [comment]
thanks for reporting. I pushed a fix for this and tagged v0.30.7
